### ISSUE TITLE
Unable to build with PHP 7.4.1 on OSX 10.14.5

### ISSUE THREAD

**Issue by lord2800** [protocolbuffers/protobuf#7041]
**What version of protobuf and what language are you using?**
Version: 3.11.2
Language: PHP

**What operating system (Linux, Windows, ...) and version?**
MacOS 10.14.5

**What runtime / compiler are you using (e.g., python version or gcc version)**
Apple LLVM 10.0.1 (clang-1001.0.46.4)

**What did you do?**
1. `brew install php@7.4` and `pecl install protobuf`

**What did you expect to see**
A successful build

**What did you see instead?**

```
$ pecl install protobuf
downloading protobuf-3.11.2.tgz ...
Starting to download protobuf-3.11.2.tgz (218,305 bytes)
.............................................done: 218,305 bytes
15 source files, building
running: phpize
Configuring for:
PHP Api Version:         20190902
Zend Module Api No:      20190902
Zend Extension Api No:   320190902
building in /private/tmp/pear/temp/pear-build-nwD8G1/protobuf-3.11.2
running: /private/tmp/pear/temp/protobuf/configure --with-php-config=/usr/local/bin/php-config
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for a sed that does not truncate output... /usr/bin/sed
checking for pkg-config... /usr/local/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-apple-darwin18.6.0
checking host system type... x86_64-apple-darwin18.6.0
checking target system type... x86_64-apple-darwin18.6.0
checking for PHP prefix... /usr/local/Cellar/php/7.4.1
checking for PHP includes... -I/usr/local/Cellar/php/7.4.1/include/php -I/usr/local/Cellar/php/7.4.1/include/php/main -I/usr/local/Cellar/php/7.4.1/include/php/TSRM -I/usr/local/Cellar/php/7.4.1/include/php/Zend -I/usr/local/Cellar/php/7.4.1/include/php/ext -I/usr/local/Cellar/php/7.4.1/include/php/ext/date/lib
checking for PHP extension directory... /usr/local/Cellar/php/7.4.1/pecl/20190902
checking for PHP installed headers prefix... /usr/local/Cellar/php/7.4.1/include/php
checking if debug is enabled... no
checking if zts is enabled... no
checking for gawk... no
checking for nawk... no
checking for awk... awk
checking if awk is broken... no
checking whether to enable Protobuf extension... yes, shared
checking for a sed that does not truncate output... /usr/bin/sed
checking for ld used by cc... /Library/Developer/CommandLineTools/usr/bin/ld
checking if the linker (/Library/Developer/CommandLineTools/usr/bin/ld) is GNU ld... no
checking for /Library/Developer/CommandLineTools/usr/bin/ld option to reload object files... -r
checking for BSD-compatible nm... /usr/bin/nm -B
checking whether ln -s works... yes
checking how to recognize dependent libraries... pass_all
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking the maximum length of command line arguments... 196608
checking command to parse /usr/bin/nm -B output from cc object... ok
checking for objdir... .libs
checking for ar... ar
checking for ranlib... ranlib
checking for strip... strip
checking for dsymutil... dsymutil
checking for nmedit... nmedit
checking for -single_module linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking if cc supports -fno-rtti -fno-exceptions... yes
checking for cc option to produce PIC... -fno-common
checking if cc PIC flag -fno-common works... yes
checking if cc static flag -static works... no
checking if cc supports -c -o file.o... yes
checking whether the cc linker (/Library/Developer/CommandLineTools/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin18.6.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no

creating libtool
appending configuration tag "CXX" to libtool
configure: patching config.h.in
configure: creating ./config.status
config.status: creating config.h
running: make
/bin/sh /private/tmp/pear/temp/pear-build-nwD8G1/protobuf-3.11.2/libtool --mode=compile cc  -I. -I/private/tmp/pear/temp/protobuf -DPHP_ATOM_INC -I/private/tmp/pear/temp/pear-build-nwD8G1/protobuf-3.11.2/include -I/private/tmp/pear/temp/pear-build-nwD8G1/protobuf-3.11.2/main -I/private/tmp/pear/temp/protobuf -I/usr/local/Cellar/php/7.4.1/include/php -I/usr/local/Cellar/php/7.4.1/include/php/main -I/usr/local/Cellar/php/7.4.1/include/php/TSRM -I/usr/local/Cellar/php/7.4.1/include/php/Zend -I/usr/local/Cellar/php/7.4.1/include/php/ext -I/usr/local/Cellar/php/7.4.1/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /private/tmp/pear/temp/protobuf/array.c -o array.lo
mkdir .libs
 cc -I. -I/private/tmp/pear/temp/protobuf -DPHP_ATOM_INC -I/private/tmp/pear/temp/pear-build-nwD8G1/protobuf-3.11.2/include -I/private/tmp/pear/temp/pear-build-nwD8G1/protobuf-3.11.2/main -I/private/tmp/pear/temp/protobuf -I/usr/local/Cellar/php/7.4.1/include/php -I/usr/local/Cellar/php/7.4.1/include/php/main -I/usr/local/Cellar/php/7.4.1/include/php/TSRM -I/usr/local/Cellar/php/7.4.1/include/php/Zend -I/usr/local/Cellar/php/7.4.1/include/php/ext -I/usr/local/Cellar/php/7.4.1/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /private/tmp/pear/temp/protobuf/array.c  -fno-common -DPIC -o .libs/array.o
In file included from /private/tmp/pear/temp/protobuf/array.c:35:
/private/tmp/pear/temp/protobuf/protobuf.h:295:72: error: unknown type name 'ulong'
static inline int php_proto_zend_hash_index_update_zval(HashTable* ht, ulong h,
                                                                       ^
/private/tmp/pear/temp/protobuf/protobuf.h:311:71: error: unknown type name 'ulong'
static inline int php_proto_zend_hash_index_update_mem(HashTable* ht, ulong h,
                                                                      ^
/private/tmp/pear/temp/protobuf/protobuf.h:340:55: error: unknown type name 'ulong'
                                                      ulong h, void** pDest) {
                                                      ^
/private/tmp/pear/temp/protobuf/protobuf.h:354:54: error: unknown type name 'ulong'
                                                     ulong h, void** pDest) {
                                                     ^
4 errors generated.
make: *** [array.lo] Error 1
ERROR: `make' failed
```
**Anything else we should know about your project / environment**

Nothing special I don't think.

**Comment by plhwin** [comment]
+1

**Comment by lucasantarella** [comment]
#7145 I just submitted a PR to fix this. I successfully got it to build with this change on my macbook.

**Comment by lord2800** [comment]
This is definitely fixed in 3.11.3.
